conans/client/plugin_manager.py

```diff
@@ -0,0 +1,89 @@
+import traceback
+import os
+import sys
+import uuid
+
+from conans.client.output import ScopedOutput
+from conans.errors import ConanException, NotFoundException
+from conans.tools import chdir
+
+
+class PluginManager(object):
+
+    def __init__(self, plugins_folder, plugin_names, output):
+        self._plugins_folder = plugins_folder
+        self._plugin_names = plugin_names
+        self.plugins = {}
+        self.output = output
+
+    def execute(self, method_name, **kwargs):
+        if not self.plugins:
+            self.load_plugins()
+
+        for name, plugin in self.plugins.items():
+            try:
+                method = getattr(plugin, method_name, None)
+                if method:
+                    output = ScopedOutput("[PLUGIN - %s] %s()" % (name, method_name), self.output)
+                    method(output, **kwargs)
+            except Exception as e:
+                raise ConanException("[PLUGIN - %s] %s(): %s\n%s" % (name, method_name, e,
+                                                                    traceback.format_exc()))
+
+    def load_plugins(self):
+        for name in self._plugin_names:
+            if name and name not in [file[:-3] for file in os.listdir(self._plugins_folder)]:
+                self.output.warn("Plugin '%s' not found in %s folder. Please remove plugin "
+                                 "from conan.conf or include it inside the plugins folder."
+                                 % (name, self._plugins_folder))
+                continue
+            plugin = self.load_plugin(name)
+            self.plugins[name] = plugin
+
+    def load_plugin(self, plugin_name):
+        filename = "%s.py" % plugin_name
+        plugin_path = os.path.join(self._plugins_folder, filename)
+        try:
+            loaded_plugin, filename = self._parse_file(plugin_path)
+            return loaded_plugin
```

Comment:
Move here the `self.plugins[name] = plugin` line from function `load_plugins`.

Also, I would catch here too the `NotFoundException` that can be raised from `_parse_file` and then move here the block with the `output.warn` that appears in function `load_plugins`.

Given these two things, the function `load_plugins` will be just a for loop...